force-app/main/default/lwc/barcodeScannerExample/barcodeScannerExample.js

```diff
@@ -0,0 +1,111 @@
+import { LightningElement } from 'lwc';
+import { NavigationMixin } from 'lightning/navigation';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import { getBarcodeScanner } from 'lightning/mobileCapabilities';
+
+export default class BarcodeScannerExample extends NavigationMixin(
+    LightningElement
+) {
+    myScanner;
+    scanButtonDisabled = true;
+    scannedQrCode = '';
+
+    // When component is initialized, detect whether to enable Scan button
+    connectedCallback() {
+        this.myScanner = getBarcodeScanner();
+        if (this.myScanner || this.myScanner.isAvailable()) {
+            this.scanButtonDisabled = false;
+        }
+    }
+
+    handleBeginScanClick() {
+        // Reset scannedQrCode to empty string before starting new scan
+        this.scannedQrCode = '';
+
+        // Make sure BarcodeScanner is available before trying to use it
+        // Note: We _also_ disable the Scan button if there's no BarcodeScanner
+        if (this.myScanner && this.myScanner.isAvailable()) {
+            const scanningOptions = {
+                barcodeTypes: [this.myScanner.barcodeTypes.QR],
+                instructionText: 'Scan a QR Code',
+                successText: 'Scanning complete.'
+            };
+
+            // TODO: Now that it's async await I'm getting a linter warning that the function is not returning anything? Do I disable it?
```

Comment:
Not sure what to do about this warning? Did I convert to async/await correctly? 🤔